### PR TITLE
[Validator] Validate readable/writable by this has an auth block

### PR DESCRIPTION
**Pull request by c-harding**
Validate that any templefile containing the metadata `#readable(by: this)`/`#writable(by: this)` uses auth, because otherwise there’s no notion of createdBy, which `by:this` relies on.

A couple of tests which this broke have been updated

### CODE REVIEW

**Review by Lewiky — APPROVED** [review]
Nice one

File: src/main/scala/temple/DSL/semantics/Validator.scala [hunk 1]
```diff
@@ -119,9 +119,13 @@ private class Validator private (templefile: Templefile) {
       case _: Metadata.ServiceLanguage => assertUnique[Metadata.ServiceLanguage]()
       case _: Metadata.Database        => assertUnique[Metadata.Database]()
       case _: Metadata.ServiceAuth     => assertUnique[Metadata.ServiceAuth]()
-      case _: Metadata.Readable        => assertUnique[Metadata.Readable]()
+      case Metadata.Readable.This if !templefile.usesAuth =>
+        errors += context.errorMessage(s"#readable(this) requires at least one service to have #auth")
+      case _: Metadata.Readable => assertUnique[Metadata.Readable]()
       case Metadata.Writable.All if metadata contains Metadata.Readable.This =>
         errors += context.errorMessage(s"#writable(all) is not compatible with #readable(this)")
```
Comment (willjvsmith) [inline src/main/scala/temple/DSL/semantics/Validator.scala:126]: Hadn't thought of this, but makes sense since you need to know ID to write

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀

**Review by jaylees14 — APPROVED** [review]
Nice one

**Pull request merged**